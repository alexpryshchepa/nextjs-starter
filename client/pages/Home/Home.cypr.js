describe('Home page', function() {
  it('Checks if default lang button is active', function() {
    cy
      .visit('/')
      .get('button')
      .contains('En')
      .should(button => {
        const className = button[0].className;
        expect(className).to.match(/^Home-langActive/);
      });
  });

  it('Switches lang', function() {
    cy
      .get('button')
      .contains('De')
      .click();

    cy
      .url()
      .should('include', '/de');

    cy
      .get('button')
      .contains('De')
      .should(button => {
        const className = button[0].className;
        expect(className).to.match(/^Home-langActive/);
      });
  });

  it('Goes to about page', function() {
    cy.get('a[href="/de/about"]').click();

    cy
      .url()
      .should('include', '/de/about');
  });
});
