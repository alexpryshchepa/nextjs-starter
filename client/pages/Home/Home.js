import React from 'react';
import PropTypes from 'prop-types';
import {
  config, i18n, withNamespaces, Link,
} from 'i18n';
import s from './Home.scss';

export const Home = ({ t }) => (
  <div className={s.root}>
    <img className={s.img} src="/static/img/nextjs.png" alt="Tech Stack" />
    <h1 className={s.title}>{t('title')}</h1>
    <Link prefetch href="/about">
      <a className={s.link}>{t('link')}</a>
    </Link>
    <p className={s.text}>{t('text')}</p>
    <ul className={s.langs}>
      {[config.defaultLanguage, ...config.otherLanguages].map((lang) => {
        const text = lang.substr(0, 1).toUpperCase() + lang.substr(1, lang.length - 1);

        return (
          <li key={lang}>
            <button
              // Issue with current lang during SSR https://github.com/isaachinman/next-i18next/issues/187
              className={i18n.language === lang ? s.langActive : s.lang}
              type="button"
              onClick={() => i18n.changeLanguage(lang)}
            >
              {text}
            </button>
          </li>
        );
      })}
    </ul>
  </div>
);

Home.getInitialProps = async () => ({
  namespacesRequired: ['common'],
});

Home.propTypes = {
  t: PropTypes.func.isRequired,
};

export default withNamespaces('common')(Home);
