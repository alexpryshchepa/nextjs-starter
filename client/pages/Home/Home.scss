@import '~client/styles/variables';
@import '~client/styles/mixins';

.root {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  padding: 64px;
  text-align: center;

  @include tablet {
    padding: 32px;
  }

  @include mobile {
    padding: 24px;
  }
}

.img {
  max-width: 100%;
  margin-bottom: 24px;
}

.title {
  font-size: 24px;
  line-height: 36px;
  margin-bottom: 24px;
}

.link {
  margin-bottom: 24px;
  color: $color-success;
}

.text {
  font-size: 14px;
  line-height: 24px;
  margin-bottom: 12px;
}

.langs {
  display: flex;
  align-items: center;
}

.lang {
  margin: 6px;
  background-color: $color-secondary;
  color: #fff;
  cursor: pointer;
}

.langActive {
  composes: lang;
  background-color: $color-primary;
}
