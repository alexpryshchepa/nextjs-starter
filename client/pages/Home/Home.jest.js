/* eslint-env jest */
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('With Enzyme', () => {
  it('Checks title text', () => {
    const component = shallow(<Home />);

    expect(
      component.find('h1[class^="Home-title"]').text(),
    ).toEqual('Welcome to NextJS starter');
  });
});

describe('With Snapshot Testing', () => {
  it('Matches with snapshot', () => {
    const component = renderer.create(<Home />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
