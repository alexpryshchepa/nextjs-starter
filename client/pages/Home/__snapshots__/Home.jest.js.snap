// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme Checks title text 1`] = `ShallowWrapper {}`;

exports[`With Enzyme Checks title text and snapshot matching 1`] = `ShallowWrapper {}`;
